android/src/main/kotlin/io/didomi/fluttersdk/EntitiesHelper.kt

```diff
@@ -0,0 +1,38 @@
+package io.didomi.fluttersdk
+import com.google.gson.Gson
+import io.didomi.sdk.Purpose
+import io.didomi.sdk.Vendor
+import java.util.*
+
+// Class used to convert Didomi entities (Vendor, Purpose) from and to types that can be sent through the Flutter channels.
+class EntitiesHelper {
+    companion object {
+        private val gson = Gson()
+        
+        // Convert a set of purposes into a list of maps.
+        @kotlin.jvm.JvmName("toListOfPurposes")
+        fun toList(purposes: Set<Purpose>): List<*> {
+            val json = gson.toJsonTree(purposes)
```

Comment:
@nicolas-chaix-didomi @pmerlet-at-didomi alternatively I could do:
```
val json = gson.toJsonTree(purposes)
val type = object : TypeToken<List<HashMap<String, *>?>?>() {}.type
return gson.fromJson(json, type)
```

Do you have any preference?

Reply:
Are there any warnings with the MR code? The current code looks more natural to me

Reply:
No warnings. As discussed, I'll keep current implementation